docs/MAINTAINING.md

```diff
@@ -28,6 +28,28 @@ http://loopback.io/doc/en/contrib/Governance.html. This means mostly:
   take a look at issues and pull requests related to code you contributed
   yourself.
 
+## Attributing your contributions
+
+Contributions of any kind welcome! We would like to acknowledge every
+contributions by the community, big or small! To do so, LoopBack 4 follows the
+[all-contributors](https://github.com/all-contributors/all-contributors)
+specification. Adding attributions is currently a manual process and we
+encourage all contributors to add their name.
+
+To add an attribution:
+
+1. Add an attribution
+
+   Use `npm run contributors:add <github-name> <contribution-type>`. This will
+   update the necessary files and create a new commit.
+
+   View the
+   [list of valid contribution types](https://allcontributors.org/docs/en/emoji-key).
+
+2. Push the changes.
+
+   Run `git push` to push the new commit.
```

Comment:
I believe our `master` branch is protected and may not allow direct pushes. I think it's better to open a pull request even for these semi-automated changes. Not only it give us a chance to run CI and review the changes, it will also trigger a notification to people watching the project. (Commits don't notify.)

Reply:
@bajtos The instructions were assuming that the readers just opened a PR, and that the workflow would be:

1. User opens a PR for a fix/feat/chore/etc.
2. User runs `contributors:add` which adds another commit
3. User pushes the new commit
4. PR gets merged, with both the original and the attribution commit.

The idea was to minimize the number of PRs and the effort it takes to merge. But we can do it the other way too:

1. User opens a PR for a fix/feat/chore/etc. (original PR)
2. Original PR gets merged
3. User checks out a new branch for adding attribution
4. User runs `contributors:add` which creates a new commit
5. User pushes the new commit and branch
6. User creates PR (attribution PR)
4. Attribution PR gets merged, with the attribution commit.

Reply:
The instructions in this section don't mention pull requests at all, that's why I assumed the intention is to commit directly to master.

> 1.  Add an attribution
> 2. Push the changes.

Can you please add more content to this section and clarify that we expect a feature branch & a pull request to be used? I think we don't need to duplicate instructions for working with feature branches and pull requests (we already have docs for that elsewhere), just mention it in the list of steps.

I am not sure if it's a good idea to ask contributors to add themselves to the list as part of their pull request contributing a feature. It adds noise to pull requests and increases chances of merge conflicts. I think it's better to update contributors on a semi-regular basis using a short-lived pull request (i.e. create a new branch, update contributors, open a new PR, wait for CI to finish, merge the PR without waiting for review).

Maybe we can support both options and explain in the documentation why/when to use which one?